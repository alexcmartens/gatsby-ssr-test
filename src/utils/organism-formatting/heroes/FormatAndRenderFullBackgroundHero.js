import React, { useContext } from "react"
import ColorSchemeConnector from "../../../components/ColorSchemeConnector/ColorSchemeConnector"
import { fullBackgroundHeroVariants } from "../../variants/fullBackgroundHeroVariants"
import { formatBackgroundImage, formatImage } from "../formatBaseMolecules"
import formatTheme from "../formatTheme"
import formatVideo from "../formatVideo"
import { FullBackgroundHero } from "terra-one"
import PageSections from "../../../components/PageSections/PageSections"
import { LayoutContext } from "../../../components/Layout/Layout"

const GetContentSections = (
  sections,
  location,
  locale,
  variant,
  homepageSlug,
  currentSpaceId,
  products,
  siteIdentifier
) => {
  if (!sections) {
    return null
  }
  const updatedSections = JSON.parse(JSON.stringify(sections))
  updatedSections.contentWidth = false
  const checkForVideosInCTA = checkForVideosInSections(updatedSections)
  const checkForProductPageLinks = checkForProductPagesInCTA(
    checkForVideosInCTA,
    homepageSlug
  )
  return (
    <PageSections
      pageSections={[checkForProductPageLinks]}
      location={location}
      locale={locale}
      isNested={true}
      marginTopAndBottom={
        variant?.variant && variant?.variant === "variant-2" ? false : true
      }
      currentSpaceId={currentSpaceId}
      products={products}
      siteIdentifier={siteIdentifier}
    />
  )
}

const checkForProductPagesInCTA = (sections, homepageSlug) => {
  if (sections?.__typename === "ContentfulCtaWithTextAndHeader") {
    sections?.ctaButtons?.map(cta => {
      if (cta?.url?.internal?.type === "ContentfulProductPage") {
        cta.url.url = `products/${cta?.url?.url}`
        return cta
      } else if (cta?.url?.url && cta?.url?.url === homepageSlug) {
        cta.url.url = ` `
        return cta
      } else {
        return cta
      }
    })
  }
  return sections
}

export const checkForVideosInSections = sections => {
  if (sections?.__typename === "ContentfulCtaWithTextAndHeader") {
    const findVideosInCTA = sections?.ctaButtons?.filter(cta => {
      return cta?.video
    })
    if (findVideosInCTA && findVideosInCTA?.length) {
      const updatedVideo = findVideosInCTA?.map(video => {
        return formatVideo(video?.video)
      })
      sections.video = updatedVideo[0]
      return sections
    } else {
      return sections
    }
  } else {
    return sections
  }
}

const FormatAndRenderFullBackgroundHero = ({
  data,
  location,
  locale,
  paddingClass,
  productPage,
  index,
  currentSpaceId,
  products,
  siteIdentifier,
}) => {
  let productPageTheme
  const { homepageSlug } = useContext(LayoutContext)

  if (productPage) {
    productPageTheme = {
      accordionIconColor: "#89C6ED",
      backgroundColor: "#0063A3",
      color: "#FFFFFF",
      ctaOne: "secondary--1",
      ctaTwo: "secondary--2",
      darkTheme: true,
      iconColor: "#FFFFFF",
      theme: "blue",
    }
  }

  const themeValue = productPageTheme
    ? productPageTheme
    : ColorSchemeConnector(data?.colorScheme)
  const theme = formatTheme(themeValue)
  const variant = fullBackgroundHeroVariants.find(option => {
    return option.option === data?.variant
  })
  const borderColorValue = ColorSchemeConnector(data?.shapeBorderColor)
  const shapeBorderColor = formatTheme(borderColorValue)

  // if FBH is first/second element on page, then don't lazyLoad it
  const lazyLoad = index < 2 ? false : true

  //assign content name for 2x2 variant css class
  const heroRightContentType = data?.rightContent?.__typename
    ? data?.rightContent?.__typename
    : null

  const fullBgBackgroundImage = data?.backgroundImage
    ? {
        ...formatBackgroundImage(data?.backgroundImage),
        lazyLoad,
      }
    : null

  const fullBgBottomImage = data?.bottomBackgroundImage
    ? {
        ...formatBackgroundImage(data?.bottomBackgroundImage),
        lazyLoad,
      }
    : null

  const content = {
    variant: productPage ? "variant-3" : variant?.variant,
    backgroundImage: fullBgBackgroundImage,
    bottomBackgroundImage: fullBgBottomImage,
    leftContent: GetContentSections(
      data?.leftContent,
      location,
      locale,
      variant,
      homepageSlug,
      currentSpaceId,
      products,
      siteIdentifier
    ),
    rightContent: GetContentSections(
      data?.rightContent,
      location,
      locale,
      variant,
      homepageSlug,
      currentSpaceId,
      products,
      siteIdentifier
    ),
    rightContentType: heroRightContentType ? heroRightContentType : null,
    rightImage: formatImage(data?.rightImage),
    shapeType: data?.shapeType?.toLowerCase(),
    shapeBorderColor: shapeBorderColor?.backgroundColor,
  }

  return (
    <FullBackgroundHero
      content={content}
      theme={theme}
      motion={data?.animation}
      paddingClass={paddingClass}
    />
  )
}

export default FormatAndRenderFullBackgroundHero
