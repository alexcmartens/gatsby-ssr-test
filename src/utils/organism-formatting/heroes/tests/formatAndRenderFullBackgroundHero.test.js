import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderFullBackgroundHero from "../FormatAndRenderFullBackgroundHero"

jest.mock("../../formatTheme")

describe("FormatAndRenderFullBackgroundHero", () => {
  it("renders correctly", () => {
    const data = {
      colorScheme: "White",
      variant: "Top Half",
      backgroundImage: {
        title: "720x720",
        image: {
          title: "720x720",
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/2aIM2EvXTK5yrpIHrvP5k5/6f7a3669abb2bf2ec39b635a91da90f6/720x720.jpg",
            details: { image: { height: 720, width: 720 } },
          },
        },
        focalPoint: { focalPoint: { y: 224, x: 358 } },
      },
      bottomBackgroundImage: {
        title: "720x720",
        image: {
          title: "720x720",
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/2aIM2EvXTK5yrpIHrvP5k5/6f7a3669abb2bf2ec39b635a91da90f6/720x720.jpg",
            details: { image: { height: 720, width: 720 } },
          },
        },
        focalPoint: { focalPoint: { y: 224, x: 358 } },
      },
      leftContent: {
        __typename: "ContentfulCtaWithTextAndHeader",
        variant: "transparent light text",
        alignment: "left",
        eyebrow: "Eyebrow Lorem ipsum",
        headline: "Headline title",
        subhead: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        body: {
          childMarkdownRemark: {
            html: "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Blandit vitae molestie aliquam iaculis eu et ultricies nec. Facilisis velit orci, amet mattis arcu pretium a. Tincidunt lacinia tempus dignissim elementum.</p>",
          },
        },
        ctaButtons: [
          {
            text: "About",
            url: "/about",
          },
          {
            text: "All Feats",
            url: "/all-feats",
          },
        ],
      },
      rightContent: null,
      rightContentType: null,
      rightImage: {
        __typename: "ContentfulImageWithFocalPoint",
        title: "720x720",
        image: {
          title: "720x720",
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/2aIM2EvXTK5yrpIHrvP5k5/6f7a3669abb2bf2ec39b635a91da90f6/720x720.jpg",
            details: { image: { height: 720, width: 720 } },
          },
        },
        focalPoint: { focalPoint: { y: 224, x: 358 } },
      },
      shapeType: "circle",
      shapeBorderColor: "Trimble blue",
      animation: false,
    }

    const tree = renderer
      .create(<FormatAndRenderFullBackgroundHero data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
