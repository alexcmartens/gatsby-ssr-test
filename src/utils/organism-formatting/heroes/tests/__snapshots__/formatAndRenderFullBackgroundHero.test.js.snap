// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderFullBackgroundHero renders correctly 1`] = `
<div
  content={
    Object {
      "backgroundImage": Object {
        "lazyLoad": true,
      },
      "bottomBackgroundImage": Object {
        "lazyLoad": true,
      },
      "leftContent": <PageSections
        isNested={true}
        marginTopAndBottom={true}
        pageSections={
          Array [
            Object {
              "__typename": "ContentfulCtaWithTextAndHeader",
              "alignment": "left",
              "body": Object {
                "childMarkdownRemark": Object {
                  "html": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Blandit vitae molestie aliquam iaculis eu et ultricies nec. Facilisis velit orci, amet mattis arcu pretium a. Tincidunt lacinia tempus dignissim elementum.</p>",
                },
              },
              "contentWidth": false,
              "ctaButtons": Array [
                Object {
                  "text": "About",
                  "url": "/about",
                },
                Object {
                  "text": "All Feats",
                  "url": "/all-feats",
                },
              ],
              "eyebrow": "Eyebrow Lorem ipsum",
              "headline": "Headline title",
              "subhead": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
              "variant": "transparent light text",
            },
          ]
        }
      />,
      "rightContent": null,
      "rightContentType": null,
      "rightImage": Object {
        "alt": "720x720",
        "focalPointImage": Object {
          "disableImageCompression": undefined,
          "height": 720,
          "title": "720x720",
          "url": "//images.ctfassets.net/citn2sn5tdjr/2aIM2EvXTK5yrpIHrvP5k5/6f7a3669abb2bf2ec39b635a91da90f6/720x720.jpg",
          "width": 720,
          "x": 358,
          "y": 224,
        },
        "type": "focalPointImage",
        "wrapperHeight": undefined,
      },
      "shapeBorderColor": "#F8F8FB",
      "shapeType": "circle",
      "variant": "variant-3",
    }
  }
  motion={false}
  theme={
    Object {
      "accordionIconColor": "#0063A3",
      "backgroundColor": "#F8F8FB",
      "color": "#242D2F",
      "ctaOne": "primary--1",
      "ctaTwo": "secondary--1",
      "iconColor": "#252A2E",
      "theme": "theme-1",
    }
  }
/>
`;
